addons/hr_expense/models/product_template.py

```diff
@@ -17,7 +17,8 @@ def create(self, vals):
             vals.update({'supplier_taxes_id': False})
         return super(ProductTemplate, self).create(vals)
 
-    @api.onchange('type')
-    def _onchange_type_for_expense(self):
-        if self.type not in ['consu', 'service']:  # storable can not be expensed.
-            self.can_be_expensed = False
+    @api.depends('type')
+    def _compute_can_be_expensed(self):
+        for product_template in self:
```

Comment:
```suggestion
        for product_template in self.filtered(lambda p: p.type not in ['consu', 'service']):
```
I think that you get it, you can apply this to the remaining diff too ;)

Reply:
You may also write in a single line
```python3
self.filtered(lambda p: p.type not in ['consu', 'service']).update({'can_be_expensed': False})
```